scripts/cluster_buildings.py

```diff
@@ -2,152 +2,101 @@
+    and adds the information to the geodataframe.
+    """
+    microgrid_buildings = gpd.read_file(input_file)
+    microgrid_buildings.rename(columns={"tags.building": "tags_building"}, inplace=True)
+    microgrid_buildings = microgrid_buildings.loc[
+        microgrid_buildings.geometry.type != "Point"
+    ]
+    microgrid_buildings = microgrid_buildings.to_crs(crs)
+    microgrid_buildings["area_m2"] = microgrid_buildings.geometry.area
+    idxs_house = microgrid_buildings.query(
+        "(tags_building == 'yes') and (area_m2 < @house_area_limit)"
+    ).index
+    microgrid_buildings.loc[idxs_house, "tags_building"] = "house"
+    return microgrid_buildings
 
 
-def get_central_points_geojson(input_filepath, output_filepath, n_clusters):
-    # Load GeoJSON data from file
-    with open(input_filepath, "r") as f:
-        data = json.load(f)
-
-    # Extract coordinates of all points
-    points = []
-    for feature in data["features"]:
-        if feature["geometry"]["type"] == "Point":
-            points.append(feature["geometry"]["coordinates"])
-
-    # Convert points to NumPy array for use with scikit-learn
-    points = np.array(points)
-
-    # Perform k-means clustering
-    kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(points)
-
-    # Compute centroids of each cluster
+def get_central_points_geojson_with_buildings(
+    input_filepath,
+    output_filepath_centroids,
+    n_clusters,
+    crs,
+    house_area_limit,
+    output_filepath_buildings,
+    output_path_csv,
+):
+    """
+    Divides the buildings into the desired number of clusters by using the kmeans function
+    and returns three different output: a geodataframe with the coordinates of the centroids of each cluster,
+    a dataframe with all of the buildings divided into clusters,
+    a csv file where for each cluster the building types are counted
+    """
+    microgrid_buildings = buildings_classification(
+        input_filepath, crs, house_area_limit
+    )
+    centroids_building = [
+        (row.geometry.centroid.x, row.geometry.centroid.y)
+        for row in microgrid_buildings.itertuples()
+    ]
+    centroids_building = np.array(centroids_building)
+    kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(centroids_building)
     centroids = kmeans.cluster_centers_
-
-    # Select most central point in each cluster
     central_points = []
+
     for i in range(kmeans.n_clusters):
-        cluster_points = points[kmeans.labels_ == i]
+        cluster_points = centroids_building[kmeans.labels_ == i]
         distances = np.linalg.norm(cluster_points - centroids[i], axis=1)
         central_point_idx = np.argmin(distances)
         central_points.append(cluster_points[central_point_idx])
-
-    # Create GeoJSON feature for each central point
-    features = []
+    central_features = []
     for i, central_point in enumerate(central_points):
```

Comment:
Also this one can be improved, but I'd be prone to keep like this and improve later.
This PR contains alread a lot of well needed features and it works :)